tests/test_mnist_lenet.py

```diff
@@ -0,0 +1,16 @@
+import tempfile
+
+from six.moves import cPickle
+
+from mnist_lenet import main
+
+
+def test_mnist_lenet():
+    with tempfile.NamedTemporaryFile() as f:
+        main(f.name, 1, True)
+        with open(f.name, "rb") as source:
+            main_loop = cPickle.load(source)
```

Comment:
Main loops serialized with the custom `dump` function can only be loaded with the custom `load` function.